context/GameContext.tsx

```diff
@@ -199,22 +166,28 @@ const gameReducer = (state: Game, action: GameAction) => {
     case "SET_LOADING":
       updatedState.board[action.payload.row].isLoading = action.payload.loading;
       return updatedState;
+    case "SET_ERROR":
+      updatedState.board[action.payload.row].isError = action.payload.isError;
+      return updatedState;
+    case "SET_CANCELLED":
+      updatedState.board[action.payload.row].isCancelled = action.payload.isCancelled;
+      return updatedState;
     case "NEW_GAME":
       const game: Game = JSON.parse(JSON.stringify(DEFAULT_GAME));
       game.id = Math.random();
       game.focusedRow = -1;
       return game;
+    default:
+      return state;
   }
 };
 
 const GameProvider: React.FC<{ children: JSX.Element }> = ({ children }) => {
   const toast = useToast();
-
-  const [game, dispatch] = React.useReducer(
-    gameReducer,
-    JSON.parse(JSON.stringify(DEFAULT_GAME))
-  );
-  const { verifying, verified, error, onVerifyProof } = useZkVerify(null);
+  const [game, dispatch] = React.useReducer(gameReducer, JSON.parse(JSON.stringify(DEFAULT_GAME)));
+  const [verifying, setVerifying] = useState(false);
```

Comment:
One improvement would be to keep track of the row that's being verified. 
`const [verifying, setVerifying] = useState('');`
Then in the UI, only one button will be in loading state and the others can be just in the disabled state. Not a blocker.

<img width="1261" alt="Screen Shot 2024-10-08 at 12 25 01 PM" src="https://github.com/user-attachments/assets/e5f2107a-3fd8-4a54-af89-51965324d3d4">